types/src/system/system_contract_type.rs

```diff
@@ -29,6 +47,44 @@ pub enum SystemContractType {
     Auction,
 }
 
+impl ToBytes for SystemContractType {
+    fn to_bytes(&self) -> Result<Vec<u8>, Error> {
```

Comment:
We should implement `fn write_bytes` and `fn to_bytes` should call write_bytes. Like here https://github.com/casper-network/casper-node/blob/f61081b83cd5bf32fefc4090158bdaa7329b7d62/node/src/types/chainspec/global_state_update.rs#L73-L77 (CC @Fraser999)